### PR TITLE
Showing profile for normal users

### PR DESCRIPTION
**Pull request by Akshi149**
# Related Issue or bug
  - Normal users should be able to view profile on clicking the author's name in the post.

Fixes: #23 

# Proposed Changes
 - Added Profileview function in views.py that takes the author name as the username and render to profileview.html that shows the profile of the same.   
- Does not show profile if logged in user and the post's author are the same.
 
# Additional Info
  - it's not showing the Harsh's profile pic since it is not available in the media/profile_pics folder. 
  
# Screenshots

 **original screenshot** 

127 0 0 1_8000_# - Google Chrome 18-08-2020 02_43_17

**Updated screenshot**

Now it redirects to ```profileview/{{ post.author }}```
127 0 0 1_8000_# - Google Chrome 18-08-2020 02_50_46

### CODE REVIEW

**Comment by harsh-9in** [comment]
@Kajol-Kumari, please review it.

**Comment by harsh-9in** [comment]
@Akshi149 plz resolve conflicts.

**Comment by Akshi149** [comment]
@harsh-9in @Kajol-Kumari please check. Hope they are resolved.

File: mysite/blog/templates/blog/profileview.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+              <!--User Details-->
+      <div class="media-body">  
+          <h2 class="account-heading">{{user.username}}</h2>
+          <p class="text-secondary">{{user.email}}</p>
+      </div>
+    </div>
+</div>
+
+{% endblock %}
```
Comment (Kajol-Kumari) [inline mysite/blog/templates/blog/profileview.html:31]: please provide a blank line at the end of the file

File: mysite/blog/urls.py [hunk 2]
```diff
@@ -3,5 +3,6 @@
 
 urlpatterns = [
     path('', views.home ,name="blog-home"),
-    path('about/',views.about,name='blog-about')
+    path('about/',views.about,name='blog-about'),
+    path('profileview/<name>',views.Profileview,name='blog-profile')
 ]
```
Comment (Kajol-Kumari) [inline mysite/blog/urls.py:8]: same here. Extra blank line

File: mysite/blog/views.py [hunk 3]
```diff
@@ -23,3 +24,17 @@ def home(request):
...
+    context={
+        'user':user,
+    }
+    if request.user!=user:
+        return render(request,'blog/profileview.html', context)
+    else:
+        context={
+            'post': post.objects.all()
+        }
+        return render(request,'blog/home.html',context)
```
Comment (Kajol-Kumari) [inline mysite/blog/views.py:40]: same. Extra blank line

**Comment by harsh-9in** [comment]
yeah @Kajol-Kumari @Akshi149  we will use already made template for that

**Comment by Akshi149** [comment]
@Kajol-Kumari can you please tell me how can I restrict the user from editing the other user's profile when I render the existing template?

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari can you please tell me how can I restrict the user from editing the other user's profile when I render the existing template?

@Akshi149 You can create flag variable that will be `true` if the user is the author of the profile and `false` when not. So you can use this flag fro showing and hiding the `edit` button.

**Comment by Akshi149** [comment]
@harsh-9in @Kajol-Kumari now please check this.

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :) @harsh-9in do look at it once!

**Comment by harsh-9in** [comment]
@Akshi149 try to update from upstream before making PR. I have already sent the commands in the slack channel.

**Pull request merged**